web-api/src/error/common.rs

```diff
@@ -153,10 +153,31 @@ pub(crate) enum InvalidDocumentSnippet {
     InvalidString(InvalidString),
     /// Input document didn't yield any snippets
     NoSnippets {},
+    /// File is not base64 encoded
+    FileNotBase64Encoded,
 }
 
 impl_application_error!(InvalidDocumentSnippet => BAD_REQUEST, INFO);
 
+/// Binary upload feature it is not available.
+#[derive(Debug, Error, Display, Serialize)]
+pub(crate) struct FileUploadNotEnabled;
+
+impl_application_error!(FileUploadNotEnabled => BAD_REQUEST, INFO);
+
+/// Content-type of the uploaded file is not supported.
+#[derive(Debug, Error, Display, Serialize)]
+pub(crate) enum InvalidBinary {
+    /// Unrecognized content-type.
+    Unrecognized,
+    /// Unsupported content-type. Found {found}
+    ContentType { found: String },
```

Comment:
:wrench:  As far as I can tell this is what "formerly was known" to be a mime-type. In which case naming it media type would be more appropriate.  

In http it's set with the `Content-Type` header because it's the media type of the content, instead of the media type of the body (which could have a content transfer encoding which has it's one media type).

And I guess http is  why tikas field is also called `Content-Type`.

<details>
<summary>Random fun fact about media types</summary>: The only reason media types where named mime types was because there was a different XML related protocol which also had media types and the authors of the MIME set of RFCs wanted to avoid confusion. But the other media type has lost any relevance so now people refer to it again as media type. But to top that off is that http didn't use the MIME specs it used their own spec which is _mosty but not exactly_ the MIME specs so if you want to nitpick http never had mime types :upside_down_face:  One of the few places where MIME and http differ is in how exactly they define media types :rofl:  Through there is somewhat of a "media type" spec by now which works forth both use cases and media types in files etc. 
</details>